resources/2023-06-resource-extensibility-product-spec.md

```diff
@@ -0,0 +1,133 @@
+    -   They want to code in a particular language E.g. :Go
+
+Today Radius enables users to get started on contributing to Radius with good-first-issues but doesn’t have a model to promote contributions further. Since the beginning of its open source launch, Dapr created the components repository and invited users to build their own components to unblock their scenarios and in turn folks contributed to the project. Learning from Dapr, Radius needs to have the extensibility points defined for contributors to interact with the project so that they can build out the things that they need to enable their scenarios and in turn the community gets benefitted with those contributions.
+
+### Customer persona(s)
+<!-- Who is the target customer? Include size/org-structure/decision makers where applicable. -->
+- Platform engineers: Platform engineers are responsible for streamlining the developer-experience for their organization by defining a set of recommended practices for application teams and provide self-service capabilities for application teams/developers to use. They are responsible for defining the infrastructure-as-code templates for the application and are the primary users of Recipes.
+
+- System Reliability Engineers (SREs) : SREs are responsible for ensuring the reliability of the applications and services that are deployed in the cloud. They are responsible for maintaining the infrastructure and ensuring that the applications are running smoothly. They maintain the infrastructure-as-code templates and provide support for the applications that are deployed. 
+
+- System integrators : System integrators are partners who help enterprises integrate proprietary services with cloud native applications. They are responsible for designing and building cloud native applications on behalf of their customers.
+
+- Developers : Developers are responsible for building the cloud native applications. They are responsible for writing the code, designing and maintaining the applications.
+
+- Open-source contributors : Cloud native open-source contributors are developers who are interested in contributing to the cloud native projects. They are responsible for building the cloud native projects and contributing to the open-source projects. They can be any of the above personas who are interested in contributing to the cloud native projects. 
+
+### Positive customer outcomes
+- Author and deploy : I can bring my own applications with custom resources/services and integrate with Radius seamlessly. I don’t have to write Go code to integrate with Radius components and can use simple spec to generate the resource definition and integrate with Radius 
+
+- Recipes for user defined types: I can create Recipes for the custom resource types and deploy via Radius 
+
+- OSS contributions: I can create and contribute Recipes for the custom resource types for the community to use.
+
+## Key scenarios
+
+### Scenario 1: Deb integrates the Budgets app with Radius
```

Comment:
not sure if this falls into this scenario or is a new one in itself, but I can see the need to keep these resource types (e.g. Plaid) hosted privately, so there might be some features/scenarios necessary for authenticating into private repos/registries to leverage these custom resource types.